InvoiceMaker/Repository/WorkDoneRepository.cs

```diff
@@ -0,0 +1,165 @@
+                        wdEndedOn = reader.GetDateTimeOffset(4);
+                    }
+                    string cClientName = reader.GetString(5);
+                    bool cIsActivated = reader.GetBoolean(6);
+                    string wtWorkTypeName = reader.GetString(7);
+                    decimal wtRate = reader.GetDecimal(8);
+                    Client client = new Client(wdClientId, cClientName, cIsActivated);
+                    WorkType workType = new WorkType(wdWorkTypeId, wtWorkTypeName, wtRate);
+
+                    if (wdEndedOn.HasValue)
+                    {
+                        return new WorkDone(wdId, client, workType, wdStartedOn, wdEndedOn.Value);
+                    }
+                    else
+                    {
+                        return new WorkDone(wdId, client, workType, wdStartedOn);
+                    }
+                }
+            }
+            return null;
+        }
+
+        public void Insert(WorkDone workDone)
+        {
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+
+                string sql = @"
+                  INSERT INTO WorkDone(ClientId, WorkTypeId, StartedOn, EndedOn)
+                  VALUES
+                  (@ClientId, @WorkTypeId, @StartedOn, @EndedOn)
+                ";
+                SqlCommand command = new SqlCommand(sql, connection);
+                command.Parameters.AddWithValue("@ClientId", workDone.ClientId);
+                command.Parameters.AddWithValue("@WorkTypeId", workDone.WorkTypeId);
+                command.Parameters.AddWithValue("@StartedOn", workDone.StartedOn);
+                if (workDone.EndedOn == null) { command.Parameters.AddWithValue("@EndedOn", DBNull.Value); }
+                else { command.Parameters.AddWithValue("@EndedOn", workDone.EndedOn); }
+                command.ExecuteNonQuery();
+            }
+        }
+
+        public void Update(WorkDone workDone)
+        {
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+
+                string sql = @"
+                  UPDATE WorkDone
+                  SET ClientId = @ClientId
+                    , WorkTypeId = @WorkTypeId
+                    , StartedOn = @StartedOn
+                    , EndedOn = @EndedOn
+                  WHERE Id = @id
+                ";
+                SqlCommand command = new SqlCommand(sql, connection);
+                command.Parameters.AddWithValue("@ClientId", workDone.ClientId);
+                command.Parameters.AddWithValue("@WorkTypeId", workDone.WorkTypeId);
+                command.Parameters.AddWithValue("@StartedOn", workDone.StartedOn);
+                if (!workDone.EndedOn.HasValue) { command.Parameters.AddWithValue("@EndedOn", DBNull.Value); }
+                else { command.Parameters.AddWithValue("@EndedOn", workDone.EndedOn); }
```

Comment:
In the method above, you did the check like this: `workDone.EndedOn == null`. I'd make this two null checks match for consistency.